### PR TITLE
Enable passing of a custom userDataDir to launcher

### LINKED ISSUE #2291: Allow custom TMP_PROFILE_DIR
Launcher kept failing when started with default user data dir on macOS Sierra.  Now this probably is an issue with chrome or directory permissions on my system.

However, I was easily able to fix this by passing a custom --user-data-dir in additionalFlags
This does double the --user-data-dir argument, and maybe we could allow for it to customizing TMP_PROFILE_DIR in configuration.

### PR DESCRIPTION
**Pull request by samccone**
* rename TMP_DIR to a more accurate name. Since this is a private
internal value, this is non-breaking.

Fixes #2291
Ref #2092

### CODE REVIEW

**Comment by samccone** [comment]
CI failure is just due to flake.

**Comment by paulirish** [comment]
✅  Confirmed it works via adding a `userDataDir` option into the https://github.com/GoogleChrome/lighthouse/blob/master/docs/readme.md#using-programmatically example.

doesn't delete if passed a custom path. clears the profile if we generated.

Nice!

**Comment by paulirish** [comment]
Tests? wdyt

**Comment by wardpeet** [comment]
would be nice to test this on appveyor 😄

**Comment by samccone** [comment]
Added some simple tests @paulirish 

to test the destroy tmp, I am going to have to do some more heavy handed refactors, let me know what you think so far before I keep going.

**Comment by paulirish** [comment]
We discussed offline:

* Don't install the `.chrome` property on the launcher instance. if anything it should be in the return object of `launch()`
* move `if (this.opts.userDataDir === undefined) {` to inside of `destroyTmp` so you can test that without faking a `close` event and a `this.chrome` interface.
* verify tmpdir is deleted when it should be and not deleted when it shouldnt be

File: chrome-launcher/package.json [hunk 1]
```diff
@@ -5,16 +5,19 @@
   "scripts": {
     "build": "tsc",
     "dev": "tsc -w",
-    "test": "mocha --reporter dot test/**/*-test.js",
+    "test": "mocha -r ts-node/register --reporter dot test/**/*-test.ts",
```
Comment (paulirish) [inline chrome-launcher/package.json]: expand to `--require`

File: chrome-launcher/package.json [hunk 2]
```diff
@@ -5,16 +5,19 @@
   "scripts": {
     "build": "tsc",
     "dev": "tsc -w",
-    "test": "mocha --reporter dot test/**/*-test.js",
+    "tests": "mocha -r ts-node/register --reporter dot test/**/*-test.ts",
```
Comment (paulirish) [inline chrome-launcher/package.json]: why remove `yarn test` ? Would prefer to continue the convention..

Comment (paulirish) [inline chrome-launcher/package.json]: can you use `--require` instead of `-r` ?

Comment (paulirish) [inline chrome-launcher/package.json]: `tests` -> `test`.  here and in run_mocha

**Comment by samccone** [comment]
updated

**Comment by samccone** [comment]
Thanks everyone for the reviews

**Pull request merged**